### PR TITLE
Changes for supporting direct intuitive touch input to navigate all OpenTyrian menus.

### PR DESCRIPTION
**Pull request by bkaindl**
Hi Pelya,

In this branch are 16 commits for you to pull. While the 1st commit is just generic to provide the option -r to build.sh for enabling the autostart of the newly installed app (which I badly needed for efficient testing before committing each change), the other changes after this one are specific to support touch input in OpenTyrian menus, to provide a nice way to edit the names of saved games and pause the game when its onPause() callback gets called (it does not block in this situation when the task list if ICS is shown in front of OpenTyrian while it still continues to run normally).

If you don't agree on some patches, just merge the ones you like, and tell me if that is the case.

Please don't merge by manually committing the changes, I guess it would become difficult to handle if you apply some patches by applying the diff yourself and committing it yourself and not just pulling the change.

If there is a problem which needs to be fixed, please ask me if I'd be able to do the fixes on a new branch from which you could them pull the remaining fixed commits.

Of course, the address bernhard.kaindl@gmx.nat used as Author of the patches is not fully correct.
It should rather be @gmx.net, but this may count just as a cheap form of spam protection ;-)

In case you run into some other issues, I guess I'd fix the .gmx.nat then as well.

If you have comments to certain lines,
I guess it's best to start a discussion on a line in the diff view like I have done on "set -eu" as an example.

Cheers,
Bernhard

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -1,4 +1,10 @@
 #!/bin/sh
+set -eu
```
Comment (bkaindl) [inline build.sh:2]: This set -eu is used to abort the script on any error. This way the "&& \" can be removed from the line ends in this script as set -eu just has the same effect.

**Comment by pelya** [comment]
A lot of changes, please give me some time to review & merge them.

Pausing audio should be done entirely inside SDL, the game in general should not care about any wonky app lifecycles or anything, so I'll merge your commit, then redo it later with some configurable callback instead of sending SDLK_PAUSE.

Your soft keyboard patch is kinda similar to the CompatibilityHacksTextInputEmulatesHwKeyboard setting inside AndroidAppSettings.cfg, which I've added just recently, I think I'll remove that option, and remove the "CallJava" passage from the name of your callback.

This change: https://github.com/bkaindl/opentyrian-android/commit/cb96c4e85fbbc6e4364ca73a387414687a5de126 - I'm specialist in dirty hacks, but this particular dirty hack will not work, because every Android phone has the Back button, including all touch-only models - they have non-removable on-screen bar with buttons, which cannot be obscured by application. Also, disabling the gamemode is kinda lame, Destruct was quite playable with on-screen buttons provided by SDL.

File: build.sh (hunk 1 above)
Comment (pelya) [inline build.sh:2]: "set -eu" is Bash-specific, it does not work with Dash - the default shell on my Ubuntu 12.04

**Comment by pelya** [comment]
I'm wrong in my last comment - it actually works okay.

**Comment by pelya** [comment]
Published it with your changes to Google Play.

**Pull request merged**